warehouse/query-core/src/main/java/datawave/query/jexl/JexlASTHelper.java

```diff
@@ -594,20 +594,28 @@ public static String deconstructIdentifier(String fieldName) {
     }
     
     /**
-     * Remove the {@link #IDENTIFIER_PREFIX} from the beginning of a fieldName if it exists
-     * 
+     * Remove the {@link #IDENTIFIER_PREFIX} from the beginning of a fieldName if it exists and remove the {@link #GROUPING_CHARACTER_SEPARATOR} from the end of
+     * a fieldName if it exists
+     *
      * @param fieldName
      * @param includeGroupingContext
      * @return
      */
     public static String deconstructIdentifier(String fieldName, boolean includeGroupingContext) {
         if (fieldName != null && fieldName.length() > 1) {
+            boolean idPrefix = fieldName.charAt(0) == IDENTIFIER_PREFIX;
+            int startIndex = idPrefix ? 1 : 0;
+            int stopIndex = idPrefix ? fieldName.length() - 1 : fieldName.length();
```

Comment:
This variable is a misnomer. it is later being used in String(fieldName.getBytes(), startIndex, stopIndex); -- stopIndex here is not an index, it is a length. As a result, I was confused by this logic. The variable should be renamed to avoid confusion.

Reply:
Updated as requested